students/esokada/lesson03/mailroom.py

```diff
@@ -0,0 +1,75 @@
+#imports
+
+import sys
+
+from statistics import mean
+
+from operator import itemgetter
+
+# data structure
+
+donors = [["Usagi", [12.5, 11.17, 130.1]], ["Ami", [17, 230, 2115]],
+["Rei", [510.1, 50]], ["Makoto", [324, 22.7]], ["Minako", [310]]]
+
+
+#functions
+
+def send_thankyou():
+    donorname = input("Please input a name:\n")
+    if donorname == "cancel":
+        return
+    if donorname == "list":
+        for i in donors:
+            print(i[0])
+        donorname = input("Please input a name:\n")
+        if donorname == "cancel":
+            return
+    donationamount = input("Please enter a donation amount:\n")
+    if donationamount == "cancel":
+        return
+    existingdonor = False
+    for d in donors:
+        if d[0] == donorname:
+            existingdonor = True
+            break
+    if existingdonor == False:
+        donors.append([donorname, []])
+    for d in donors:
+        if d[0] == donorname:
+            d[1].append(float(donationamount))
+    print(f"Dear {donorname}, thank you for your generous donation of ${float(donationamount):.2f}.\n")
+
+def write_report():
+    titles = ["Donor Name", "Total Given", "Num Gifts", "Average Gift"]
+    print(f"{titles[0]:15} {titles[1]:>20} {titles[2]:>10} {titles[3]:>20}")
+    reportlines = []
+    for d in donors:
+        dname = d[0]
+        dtotal = sum(d[1])
+        dnum = len(d[1])
+        davggift = mean(d[1])
+        reportlines.append([dname, dtotal, dnum, davggift])
+    for r in sorted(reportlines, key = itemgetter(1), reverse = True):
```

Comment:
no spaces around keyword items -> `key=itemgetter(1)`

and nice use of `itemgetter` !